packages/snaps-execution-environments/src/common/endowments/endowmentModules.ava.test.ts

```diff
@@ -0,0 +1,188 @@
+});
+
+const modules = [
+  {
+    name: 'AbortController',
+    exportedNames: ['AbortController'],
+    moduleFactory: abortController,
+  },
+  {
+    name: 'AbortSignal',
+    exportedNames: ['AbortSignal'],
+    moduleFactory: abortSignal,
+  },
+  {
+    name: 'ArrayBuffer',
+    exportedNames: ['ArrayBuffer'],
+    moduleFactory: arrayBuffer,
+  },
+  {
+    name: 'atob',
+    exportedNames: ['atob'],
+    moduleFactory: atobEndowment,
+  },
+  {
+    name: 'BigInt',
+    exportedNames: ['BigInt'],
+    moduleFactory: bigInt,
+  },
+  {
+    name: 'BigInt64Array',
+    exportedNames: ['BigInt64Array'],
+    moduleFactory: bigInt64Array,
+  },
+  {
+    name: 'BigUint64Array',
+    exportedNames: ['BigUint64Array'],
+    moduleFactory: bigUint64Array,
+  },
+  {
+    name: 'btoa',
+    exportedNames: ['btoa'],
+    moduleFactory: btoaEndowment,
+  },
+  {
+    name: 'crypto',
+    exportedNames: ['crypto', 'SubtleCrypto'],
+    moduleFactory: crypto,
+  },
+  {
+    name: 'DataView',
+    exportedNames: ['DataView'],
+    moduleFactory: dataView,
+  },
+  {
+    name: 'Float32Array',
+    exportedNames: ['Float32Array'],
+    moduleFactory: float32Array,
+  },
+  {
+    name: 'Float64Array',
+    exportedNames: ['Float64Array'],
+    moduleFactory: float64Array,
+  },
+  {
+    name: 'Int8Array',
+    exportedNames: ['Int8Array'],
+    moduleFactory: int8Array,
+  },
+  {
+    name: 'Int16Array',
+    exportedNames: ['Int16Array'],
+    moduleFactory: int16Array,
+  },
+  {
+    name: 'Int32Array',
+    exportedNames: ['Int32Array'],
+    moduleFactory: int32Array,
+  },
+  {
+    name: 'Interval',
+    exportedNames: ['setInterval', 'clearInterval'],
+    moduleFactory: interval,
+  },
+  {
+    name: 'Math',
+    exportedNames: ['Math'],
+    moduleFactory: math,
+  },
+  {
+    name: 'Network',
+    exportedNames: ['fetch', 'WebSocket'],
+    moduleFactory: network,
+  },
+  {
+    name: 'TextDecoder',
+    exportedNames: ['TextDecoder'],
+    moduleFactory: textDecoder,
+  },
+  {
+    name: 'TextEncoder',
+    exportedNames: ['TextEncoder'],
+    moduleFactory: textEncoder,
+  },
+  {
+    name: 'Timeout',
+    exportedNames: ['setTimeout', 'clearTimeout'],
+    moduleFactory: timeout,
+  },
+  {
+    name: 'Uint8Array',
+    exportedNames: ['Uint8Array'],
+    moduleFactory: uint8array,
+  },
+  {
+    name: 'Uint8ClampedArray',
+    exportedNames: ['Uint8ClampedArray'],
+    moduleFactory: uint8ClampedArray,
+  },
+  {
+    name: 'Uint16Array',
+    exportedNames: ['Uint16Array'],
+    moduleFactory: uint16Array,
+  },
+  {
+    name: 'Uint32Array',
+    exportedNames: ['Uint32Array'],
+    moduleFactory: uint32Array,
+  },
+  {
+    name: 'URL',
+    exportedNames: ['URL'],
+    moduleFactory: url,
+  },
+  {
+    name: 'WebAssembly',
+    exportedNames: ['WebAssembly'],
+    moduleFactory: webAssembly,
+  },
+];
+
+modules.forEach((module) => {
+  test(`${module.name} endowment module has expected properties`, (expect) => {
```

Comment:
I know I made you rearrange this test once already, but now that I look at it, it is testing whether the fields that were once exported are the same, so it's a perfect usecase of a snapshot.

You just need to put all the imported stuff in one object and snapshot it in a single test to acheive the same thing you did here.

Reply:
Pushed the snapshot test